Sources/GRPC/ServerCallContexts/UnaryResponseCallContext.swift

```diff
@@ -69,29 +69,25 @@ open class UnaryResponseCallContextImpl<ResponsePayload>: UnaryResponseCallConte
     super.init(eventLoop: channel.eventLoop, request: request, logger: logger)
 
     responsePromise.futureResult
-      // Send the response provided to the promise.
-      .map { responseMessage -> EventLoopFuture<Void> in
-        return self.channel.writeAndFlush(NIOAny(WrappedResponse.message(.init(responseMessage, compressed: self.compressionEnabled))))
```

Comment:
This change here may be non-obvious: I actually missed it the first time I looked at this code.

While the `Future` returned from this `writeAndFlush` _looks_ important (it's even called out in the type signature of the closure), it actually isn't. This is a `map` block, not a `flatMap`, so the rest of the callback chain does not wait for the future to complete before proceeding. In fact, the very next `map` block discards the `Future` that this code went to so much effort to construct.

This is why I was able to flatten all of this code down: in the end, everything was secretly just synchronous code, hidden in an async chain.